applications/nrf_desktop/src/modules/config.c

```diff
@@ -163,36 +206,19 @@ static int enable_settings(void)
 
 		err = settings_register(&sh);
 		if (err) {
-			LOG_ERR("Cannot register settings handler");
-			goto error;
+			LOG_ERR("Cannot register settings handler: %d", err);
+		} else {
+			if (IS_ENABLED(CONFIG_DESKTOP_SETTINGS_LOAD_BY_THREAD_ENABLE)) {
+				start_loading_thread();
+			} else {
+				err = load_settings();
+			}
 		}
 	}
-
-	/* This module loads settings for all application modules */
-	err = settings_load();
-	if (err) {
-		LOG_ERR("Cannot load settings");
-		goto error;
-	}
-
-	LOG_INF("Settings loaded");
-
-error:
 	return err;
```

Comment:
return value of init_settings is not checked so there is no point for it to return int.

instead it would be good to add
```
if (err) {
   module_set_state(MODULE_STATE_ERROR);
}
```

Reply:
changed